perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,288 @@
+  ]
+  for cmd in cmds:
+    vm.RemoteCommand(cmd, should_log=True)
+
+
+def Run(benchmark_spec):
+  """Run SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Running SciMark2 on %s', vm)
+  samples = []
+  # Run the Java and C benchmarks twice each, once with defaults and
+  # once with the "-large" flag to use a larger working set size.
+  #
+  # Since the default output is not very parsing-friendly, print an
+  # extra header to identify the tests. This must match
+  # RESULT_START_REGEX as used below.
+  cmds = [
+      '(echo ";;; Java small"; cd {0} && java -cp {1} {2})'.format(
+          SCIMARK2_PATH, SCIMARK2_JAVA_JAR, SCIMARK2_JAVA_MAIN),
+      '(echo ";;; C small"; cd {0} && ./scimark2)'.format(
+          SCIMARK2_C_SRC),
+      '(echo ";;; Java large"; cd {0} && java -cp {1} {2} -large)'.format(
+          SCIMARK2_PATH, SCIMARK2_JAVA_JAR, SCIMARK2_JAVA_MAIN),
+      '(echo ";;; C large"; cd {0} && ./scimark2 -large)'.format(
+          SCIMARK2_C_SRC),
+  ]
+  for cmd in cmds:
+    stdout, _ = vm.RemoteCommand(cmd, should_log=True)
+    samples.extend(ParseResults(stdout))
+  return samples
+
+
+def Cleanup(benchmark_spec):
+  """Cleanup SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  pass
+
+
+def ParseResults(results):
+  """Result parser for SciMark2.
+
+  Sample Results (C version):
+    **                                                              **
+    ** SciMark2 Numeric Benchmark, see http://math.nist.gov/scimark **
+    ** for details. (Results can be submitted to pozo@nist.gov)     **
+    **                                                              **
+    Using       2.00 seconds min time per kenel.
+    Composite Score:         1596.04
+    FFT             Mflops:  1568.64    (N=1024)
+    SOR             Mflops:  1039.98    (100 x 100)
+    MonteCarlo:     Mflops:   497.64
+    Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+    LU              Mflops:  2899.56    (M=100, N=100)
+
+  (Yes, "kenel" is part of the original output.)
+
+  Sample Results (Java version):
+
+    SciMark 2.0a
+
+    Composite Score: 1731.4467627163242
+    FFT (1024): 996.9938397943672
+    SOR (100x100):   1333.5328291027124
+    Monte Carlo : 724.5221517116782
+    Sparse matmult (N=1000, nz=5000): 1488.18620413327
+    LU (100x100): 4113.998788839592
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_REGEX = re.compile(r'^;;; \s+ (.*)', re.X | re.M)
+
+  SCORE_REGEX = re.compile(r'''
```

Comment:
[Prefer double quotes](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Strings#Strings) for multi-line strings.